src/marten/cli/generator/auth/templates/app/schemas/password_update_schema.cr.ecr

```diff
@@ -0,0 +1,24 @@
+module <%= context.module_name %>
+  class PasswordUpdateSchema < Marten::Schema
+    setter :email
+    getter :user
+    field :old_password, :string, max_size: 128, strip: false
+    field :new_password, :string, max_size: 128, strip: false
+    field :new_confirm_password, :string, max_size: 128, strip: false
+
+    @email : String?
+    @user : User?
+
+    validate :validate_password
+
+    private def validate_password
+      return unless old_password? && new_password? && new_confirm_password?
+
+      @user = MartenAuth.authenticate(@email.not_nil!, old_password!)
```

Comment:
This is where I would leverage the already logged in user record and `User#check_password` in order to verify the old password:

```crystal
if !@user.check_password(old_password!)
  errors.add(:old_password, "Invalid user password")
end
```